### PR TITLE
Setup repo

### PR DESCRIPTION
**Pull request by shamir92**
## Summary by CodeRabbit

- **New Features**
  - Introduced JWT configuration management for token handling.
  - Added structures for medical records, patients, and users.
  - Initialized Fiber server with routing and middleware configurations.
  - Implemented Docker Compose exclusions for build context.
  
- **Documentation**
  - Added README files detailing project structure and API routing guidelines.
  - Provided Docker Compose documentation for PostgreSQL setup.

- **Bug Fixes**
  - N/A

- **Refactor**
  - N/A

- **Style**
  - N/A

- **Tests**
  - N/A

- **Chores**
  - N/A

- **Revert**
  - N/A

### CODE REVIEW

**Review by 14mdzk — APPROVED** [review]
LGTM

File: w3/configuration/jwt_configuration.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func (c *jwtConfiguration) GetIssuer() string {
+	return c.issuer
+}
+
+func (c *jwtConfiguration) getExpirationTIme() time.Time {
+	expiresInMinute, _ := strconv.ParseInt(c.expiresInMinute, 10, 64)
+
+	return time.Now().Add(time.Minute * time.Duration(expiresInMinute))
```
[the remark this answers is not in the document]
Comment (wandyirawan) [reply w3/configuration/jwt_configuration.go:40]: log fatal setahu saya akan bikin program berhenti, apakah parse jika error bikin app berhenti?

**Review by enjaytarigan — APPROVED** [review]
LGTM

File: w3/protocol/api/controller/ping_controller.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		pingUsecase: pingUsecase,
+	}
+}
+
+func (pc *pingController) GetPingController(c *fiber.Ctx) error {
+	value, _ := pc.pingUsecase.GetPing()
+	if !value {
+		return c.Status(http.StatusServiceUnavailable).JSON("pong")
+	}
+	return c.Status(http.StatusOK).JSON("pong")
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/protocol/api/controller/ping_controller.go:30]: i do not want to return the error of ping to frontend. 
so just want to ignore first.

File: w3/protocol/api/controller/ping_controller.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+type pingController struct {
+	pingUsecase usecase.IPingUsecase
+}
+
+// TODO: add all function under ping controller to inferface. this will make it easier to test
+type IPingController interface {
+	GetPingController(c *fiber.Ctx) error
+}
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/protocol/api/controller/ping_controller.go:17]: can try to open github issue.

File: w3/configuration/jwt_configuration.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+type jwtConfiguration struct {
+	signingKey      string
+	issuer          string
+	expiresInMinute string
+}
+
+type IJWTConfiguration interface {
+	GetSigningKey() string
+	GetIssuer() string
+	getExpirationTIme() time.Time
```
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/configuration/jwt_configuration.go:18]: done

File: w3/configuration/jwt_configuration.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shamir92) [reply w3/configuration/jwt_configuration.go:40]: done

**Pull request merged**